src/feature/hs_common/shared_random_client.c

```diff
@@ -31,6 +32,24 @@ srv_to_control_string(const sr_srv_t *srv)
   return srv_str;
 }
 
+/**
+ * If we have no consensus and we are not an authority, assume that this is
+ * the voting interval.  We should never actually use this: only authorities
+ * should be trying to figure out the schedule when they don't have a
+ * consensus.
+ **/
+#define DEFAULT_NETWORK_VOTING_INTERVAL (3600)
+
+/* This is an unpleasing workaround for tests.  Our unit tests assume that we
```

Comment:
Hmm agreed that this is suboptimal. I wonder if another approach would be to mock `authdir_mode()` in the relevant tests, since this macro is always used next to `authdir_mode()`.

This way it would be ultimately true that only dirauths can get into `voting_schedule.c` in a way that we could also soft-assert against it?

Reply:
We could mock dirauth mode or set V3AuthoritativeDirectory here, but the problem is how pervasive in the tests it would need to be.  All of our consensus-generation tests need to reach this code, but most of them don't actually set dirauth mode.

Without this kludge, there are 28 tests that fail and would need to get tweaked.  To me that seemed higher than I thought it was reasonable to deal with, but I could try to tweak them all anyway.  Do you think I should go ahead with that?

Reply:
(Also, it's not just authority-only functions. For example, consider the test `hs_service/intro2_handling`. It tries to get a subcredential, so it calls into the shared_random_client.c code.  In reality that would never happen without a consensus, but in our test it happens without a consensus, and so would trigger a soft assertion.)

Reply:
ACK makes sense. Let's leave it as is then.